### DISCUSSION TITLE
Internal Service Error w/ SSO preview links

### DISCUSSION THREAD

**Discussion by kiyarose** [vercel/community#7257] [Help]
### Summary

I've been using Vercel for a bit now, and randomly today I always get a serverless function error when using a Vercel Preview link...
This is happening to all of my projects on the vercel.com/sso-api endpoint. (Image below of error)
![image](https://github.com/vercel/community/assets/75678535/d59319b5-88b7-4ded-b2ca-656ce605806c)
I am coming to believe that this is possibly a Vercel error (despite what the error says), because it is happening all of a sudden on all of my projects without any other warning. 

### Steps to Reproduce

I am experiencing this on any Vercel preview link I try to use, on both production and preview deployments. 
Assigned domains are working fine, but preview links using the SSO API are not, which leads me to believe a Vercel issue (even if localized). 
I am quite at a loss of where to go next at this point, so I appreciate any info that I get. 
Apologies if I formatted or wrote something weird :)

**Comment by versecafe** [comment]
@dswan36 SSO APIs usually required explicitly allowed callback URLs and since the preview URLs have random prefixes they'll always fail, you can hit through something like an auth proxy site just to keep the callback URL the same to avoid the issues, just make sure you take care of any security risks.

**Reply by kiyarose** [reply]
> @dswan36 SSO APIs usually required explicitly allowed callback URLs and since the preview URLs have random prefixes they'll always fail, you can hit through something like an auth proxy site just to keep the callback URL the same to avoid the issues, just make sure you take care of any security risks.

@versecafe,

My apps don't specifically take advantage of Vercel's SSO in the code I have written, most of my stuff is 99.9% frontend. 
I originally thought it might have been a build error because the project I first noticed this on always flooded the logs with errors, but the prod build was successful on the normal vercel.app domain and the added custom domain. 

I noticed this happening on some of my other projects (the preview URLs not working but prod URLs working fine). 
This seems to happen when Vercel tries to verify my identity to access the preview link. As I always error out @ https://vercel.com/sso or @ https://vercel.com/oauth/git?scope=sso-api . I was considering it a server side deployment issue at first, but considering I haven't seen myself any other issue reports, and Vercel itself seems to not consider it their side (Error message, nothing out of the ordinary on the status website  [despite the error happening on a Vercel controlled page]). 

If I didn't know any better I would consider it a shard failure, but considering the lack of an influx of issue reports, I wouldn't know. 
Some of my preview URLs are working nominally at the moment, but randomly when authing, it fails with the serverless function error on the SSO page. 

Considering I haven't had issues before, and not much of significance could have changed in my code. Plus no other failure reports (To my knowledge). 
I find it peculiar.

[a comment by an automated account is not shown]
